File: assignment-test(java)/src/main/java/com/assignment/assignmenttest/service/RandomUserService.java
package com.assignment.assignmenttest.service;

import com.assignment.assignmenttest.model.RandomUserResponse;
import com.assignment.assignmenttest.webclient.RandomUserClient;
import com.assignment.assignmenttest.webclient.model.NameModel;
import com.assignment.assignmenttest.webclient.model.RandomUserClientResponse;
import com.fasterxml.jackson.core.JsonProcessingException;
import org.springframework.stereotype.Service;

@Service
public class RandomUserService {

    private final RandomUserClient randomUserClient;

    public RandomUserService(RandomUserClient randomUserClient) {
        this.randomUserClient = randomUserClient;
    }

    public RandomUserResponse getUser(String seed){
        RandomUserClientResponse responseClient = new RandomUserClientResponse();
        try {
            responseClient = randomUserClient.callRandomUser(seed);
        } catch (JsonProcessingException e) {
            throw new RuntimeException(e);
        }
        return mappingDataRandomUser(responseClient);
    }

    public RandomUserResponse mappingDataRandomUser(RandomUserClientResponse randomUserClientResponse){
        RandomUserResponse randomUserResponse = new RandomUserResponse();
        var result = randomUserClientResponse.getResults().stream().findFirst();
        if(result.isPresent()){
            NameModel nameModel= result.get().getName();
            if(nameModel!=null) {
                randomUserResponse.setFirstName(nameModel.getFirst());
                randomUserResponse.setLastName(nameModel.getLast());
            }
            randomUserResponse.setGender(result.get().getGender());
            randomUserResponse.setEmail(result.get().getEmail());
        }
        return randomUserResponse;
    }
}

File: assignment-test(kotlin)/src/main/kotlin/com/assignment/test/service/RandomUserService.kt
package com.assignment.test.service

import com.assignment.test.service.model.RandomUserResponse
import com.assignment.test.webclients.RandomUserClient
import com.assignment.test.webclients.model.ResultModel
import org.springframework.stereotype.Service

@Service
class RandomUserService (val randomUserClient: RandomUserClient){

     fun getUser(seed : String): RandomUserResponse? {
        val responseClient = randomUserClient.callRandomUser(seed)?.results?.firstOrNull();
        return responseClient?.let { mappingDataRandomUser(it) }
    }

    private fun mappingDataRandomUser(randomUserClientResponse: ResultModel)=
            randomUserClientResponse.let{ s->
                RandomUserResponse(firstname = s.name?.first.orEmpty(),
                        lastname = s.name?.last.orEmpty(),
                        gender = s.gender.orEmpty(),
                        email = s.email.orEmpty())
            }
}